src/driver/cdev_sgdma.c

```diff
@@ -300,17 +298,18 @@ static ssize_t char_sgdma_read_write(struct file *file, const char __user *buf,
 	return res;
 }
 
-
 static ssize_t char_sgdma_write(struct file *file, const char __user *buf,
 				size_t count, loff_t *pos)
 {
-	return char_sgdma_read_write(file, (char *)buf, count, pos, 1);
+	return char_sgdma_read_write(file->private_data, (char *)buf, count,
+			pos, 1);
```

Comment:
The point of rewriting the driver is so that there's a single unified interface. We should only have one char device.

So you should probably be moving the functionality you need and removing this file. Or simply changing this file to be more what you need.

Reply:
There is a single unified interface, I kept this particular char device around for debugging/testing and made it optional with  8dbca64a9b0267157e0407c982648eb36034bd59. But I don't feel too strongly about it, so I'll remove it in the next version.